processor/transform/mdxish/normalize-malformed-md-syntax.ts

```diff
@@ -0,0 +1,165 @@
+import type { Emphasis, Parent, Root, Strong, Text } from 'mdast';
+import type { Plugin } from 'unified';
+
+import { SKIP, visit } from 'unist-util-visit';
+
+/**
+ * A remark plugin that normalizes malformed bold and italic markers in text nodes.
+ * Detects patterns like `** bold**`, `Hello** Wrong Bold**`, `__ bold__`, `Hello__ Wrong Bold__`,
+ * `* italic*`, `Hello* Wrong Italic*`, `_ italic_`, or `Hello_ Wrong Italic_`
+ * and converts them to proper strong/emphasis nodes, matching the behavior of the legacy rdmd engine.
+ *
+ * Supports both asterisk (`**bold**`, `*italic*`) and underscore (`__bold__`, `_italic_`) syntax.
+ * Also supports snake_case content like `** some_snake_case**`.
+ *
+ * This runs after remark-parse, which (in v11+) is strict and doesn't parse
+ * malformed emphasis syntax. This plugin post-processes the AST to handle these cases.
+ */
+const normalizeEmphasisAST: Plugin = () => (tree: Root) => {
+  visit(tree, 'text', function visitor(node: Text, index, parent: Parent) {
+    if (index === undefined || !parent) return undefined;
+
+    // Skip if inside code blocks or inline code
+    if (parent.type === 'inlineCode' || parent.type === 'code') {
+      return undefined;
+    }
+
+    const text = node.value;
+
+    // Patterns to detect for bold (** and __) and italic (* and _) syntax:
+    // Bold: ** text**, **text **, word** text**, ** text **
+    // Italic: * text*, *text *, word* text*, * text *
+    // Same patterns for underscore variants
+    // We use separate patterns for each marker type to allow this flexibility.
+
+    // Pattern for ** bold **
+    // Groups: 1=wordBefore, 2=marker, 3=contentWithSpaceAfter, 4=trailingSpace1, 5=contentWithSpaceBefore, 6=trailingSpace2, 7=afterChar
+    // trailingSpace1 is for "** text **" pattern, trailingSpace2 is for "**text **" pattern
+    const asteriskBoldRegex =
+      /([^*\s]+)?\s*(\*\*)(?:\s+((?:[^*\n]|\*(?!\*))+?)(\s*)\2|((?:[^*\n]|\*(?!\*))+?)(\s+)\2)(\S|$)?/g;
+
+    // Pattern for __ bold __
+    const underscoreBoldRegex =
+      /([^_\s]+)?\s*(__)(?:\s+((?:[^_\n]|_(?!_))+?)(\s*)\2|((?:[^_\n]|_(?!_))+?)(\s+)\2)(\S|$)?/g;
+
+    // Pattern for * italic *
+    const asteriskItalicRegex = /([^*\s]+)?\s*(\*)(?!\*)(?:\s+([^*\n]+?)(\s*)\2|([^*\n]+?)(\s+)\2)(\S|$)?/g;
+
+    // Pattern for _ italic _
+    const underscoreItalicRegex = /([^_\s]+)?\s*(_)(?!_)(?:\s+([^_\n]+?)(\s*)\2|([^_\n]+?)(\s+)\2)(\S|$)?/g;
```

Comment:
Can you move these regex definitions up to the module scope. Defining them in the visitor function means they are recreated for every node in the AST right?